---
title: "README"
author: "Anna Hutchinson"
output: md_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(cowplot)
library(dplyr)
```

---

### Aim

---

Investigate the utility of incorporating functional annotation data for fine-mapping causal variants using PAINTOR.

---

### Method

---

Simulate loci that reach genome-wide significance and corresponding binary annotation vectors, where 5% of the SNPs are randomly allocated the annotation.

Run PAINTOR on various combinations of the simulated loci, where the proportion of CVs with the annotation varies across combinations. The annotation is uninformative in combinations where the annotation is present in 5% of the CVs (as this is its proportion genome-wide) and becomes more informative as the number of CVs with the annotation increases in that specific combination of simulated loci.

---

### Code

---

The code to run this analysis is avaliable in the /code directory of this github repository. The files should be run in the following order:

1. `simulate_loci.R`

2. `make_loci_combinations.R`

3. `run_PAINTOR`

4. `make_finalres.R`

---

### Results

---

```{r warning = FALSE, message=FALSE, comment = FALSE, echo = FALSE, fig.width = 10, fig.height = 7}
x <- readRDS("res/fullres.RDS")

x$prop <- as.factor(x$prop)

x <- cbind(x, bin = cut(x$max_Z, quantile(x$max_Z, probs=0:5/5), include.lowest=TRUE))

quantiles_low <- x[which(x$ld == "low"),]

quantiles_medium <- x[which(x$ld == "medium"),]

quantiles_high <- x[which(x$ld == "high"),]

new.df_low <- quantiles_low %>%
  group_by(prop, bin) %>%
  summarise(PP_mean = mean(PPs))

new.df_medium <- quantiles_medium %>%
  group_by(prop, bin) %>%
  summarise(PP_mean = mean(PPs))

new.df_high <- quantiles_high %>%
  group_by(prop, bin) %>%
  summarise(PP_mean = mean(PPs))

low_mean <- ggplot(new.df_low, aes(x = prop, y = PP_mean, col = bin, group = bin))+geom_line()+ theme_cowplot(12) +theme(plot.title = element_text(color="black", face="bold", size = 12, hjust=0.5)) + background_grid(major = "xy", minor = "none") + ggtitle("Low LD")+ ylab("Mean PP at CV")+ theme(legend.position = "none") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + xlab("Proportion of CVs with annotation")

medium_mean <- ggplot(new.df_medium, aes(x = prop, y = PP_mean, col = bin, group = bin))+geom_line()+ theme_cowplot(12) +theme(plot.title = element_text(color="black", face="bold", size = 12, hjust=0.5), legend.text=element_text(size=11.5)) + background_grid(major = "xy", minor = "none") + ggtitle("Medium LD")+ ylab("") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + xlab("Proportion of CVs with annotation") + scale_colour_discrete(name = "Maximum absolute\nZ score")

high_mean <- ggplot(new.df_high, aes(x = prop, y = PP_mean, col = bin, group = bin))+geom_line()+ theme_cowplot(12) +theme(plot.title = element_text(color="black", face="bold", size = 12, hjust=0.5)) + background_grid(major = "xy", minor = "none") + ggtitle("High LD")+ ylab("")+ theme(legend.position = "none") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + xlab("Proportion of CVs with annotation")

quantile_plot <- plot_grid(low_mean + coord_cartesian(ylim = c(0,1)), medium_mean + theme(legend.position = "none")+ coord_cartesian(ylim = c(0,1)), high_mean+ coord_cartesian(ylim = c(0,1)), nrow = 1)

legend <- get_legend(
  # create some space to the left of the legend
  medium_mean + theme(legend.box.margin = margin(0, 0, 0, 10))
)

final <- plot_grid(quantile_plot, legend, rel_widths = c(3, .5))

final

# ggsave("functional-annots.png",plot=final,height=8,width=12)
```

---